packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/settings/PluginSettings.kt

```diff
@@ -0,0 +1,46 @@
+/**
+ * Settings state for the plugin. These classes are responsible for the persistence of the plugin
+ * settings.
```

Comment:
How "broad" is the scope of settings in IntelliJ? Is every setting defined here automatically visible and configurable to the user, or is it more like Compass, where it's fine for us to use some settings as easy ways of persisting small bits of data?

Reply:
It's more like Compass. So you have the "settings state" that is persisted and it's not related anyhow to the settings view. How you want to handle these settings is up to you. There is a class (com.mongodb.jbplugin.settings.PluginSettingsConfigurable) that specifies how this settings can be configured in the Settings UI and it's not mandatory to add all of them, only whichever are relevant. 

To give you an example, `hasTelemetryOptOutputNotificationBeenShown` is not set up in the Settings page, but is set up in the initial activity (com.mongodb.jbplugin.ActivatePluginPostStartupActivity).